meilisearch/tests/index/create_index.rs

```diff
@@ -34,7 +34,7 @@ async fn create_index_with_gzip_encoded_request() {
 
     assert_eq!(response["status"], "enqueued");
 
-    let response = index.wait_task(0).await;
+    let response = index.wait_task(response.uid()).await;
```

Comment:
Same as above, we could use a shared server.

But this time we don’t have the right function to do it with an encoder.
Could you create a new method on the `Server<Shared>` called `unique_index_with_encoder` that takes the encoder by parameter?

Here’s the current one you need to duplicate:
https://github.com/meilisearch/meilisearch/blob/7f20c13f3fc374faa0b812d4e21de541ed95746a/meilisearch/tests/common/server.rs#L301-L310